change test for a warning to filterwarnings since the code may or not issue a warning depending on an arch
see https://github.com/scipy/scipy/issues/17954 + https://github.com/scipy/scipy/pull/17975
diff --git a/scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py b/scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py
index afd7a106ea0..87f3756c37d 100644
--- a/scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py
+++ b/scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py
@@ -255,17 +255,18 @@
     _check_fiedler(12, 2)
 
 
-@pytest.mark.skipif(platform.machine() == 'aarch64',
-                    reason="issue #15935")
+@pytest.mark.filterwarnings("ignore:Failed at iteration")
+@pytest.mark.filterwarnings("ignore:Exited at iteration")
+@pytest.mark.filterwarnings("ignore:Exited postprocessing")
 def test_failure_to_run_iterations():
     """Check that the code exists gracefully without breaking. Issue #10974.
+    The code may or not issue a warning, filtered out. Issue #15935, #17954.
     """
     rnd = np.random.RandomState(0)
     X = rnd.standard_normal((100, 10))
     A = X @ X.T
     Q = rnd.standard_normal((X.shape[0], 4))
-    with pytest.warns(UserWarning, match="Failed at iteration"):
-        eigenvalues, _ = lobpcg(A, Q, maxiter=40, tol=1e-12)
+    eigenvalues, _ = lobpcg(A, Q, maxiter=40, tol=1e-12)
     assert(np.max(eigenvalues) > 0)
 
 
